Snippets/ServiceFabricPersistence/ServiceFabricPersistence_1/SagasConfiguration.cs

```diff
@@ -0,0 +1,83 @@
+﻿namespace ServiceFabricPersistence_1
+{
+    using System.Globalization;
+    using System.IO;
+    using System.Threading.Tasks;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Converters;
+    using NServiceBus;
+    using NServiceBus.Persistence.ServiceFabric;
+
+    public class SagasConfiguration
+    {
+        public void ConfigureCustomJsonSerializerSettings(EndpointConfiguration endpointConfiguration)
+        {
+            #region ServiceFabricPersistenceSagaJsonSerializerSettings
+            var persistence = endpointConfiguration.UsePersistence<ServiceFabricPersistence>();
+
+            persistence.SagaSettings().JsonSettings(new JsonSerializerSettings
+            {
+                TypeNameHandling = TypeNameHandling.Auto,
+                Converters =
+                {
+                    new IsoDateTimeConverter
+                    {
+                        DateTimeStyles = DateTimeStyles.RoundtripKind
+                    }
+                }
+            });
+            #endregion
+        }
+
+        public void ConfigureCustomJsonReader(EndpointConfiguration endpointConfiguration)
+        {
+            #region ServiceFabricPersistenceSagaReaderCreator
+            var persistence = endpointConfiguration.UsePersistence<ServiceFabricPersistence>();
+
+            persistence.SagaSettings().ReaderCreator(textReader => { return new JsonTextReader(textReader); });
+            #endregion
+        }
+
+        public void ConfigureCustomJsonWriter(EndpointConfiguration endpointConfiguration)
+        {
+            #region ServiceFabricPersistenceSagaWriterCreator
+            var persistence = endpointConfiguration.UsePersistence<ServiceFabricPersistence>();
+
+            persistence.SagaSettings().WriterCreator(builder =>
+            {
+                var writer = new StringWriter(builder);
+                return new JsonTextWriter(writer)
+                {
+                    Formatting = Formatting.None
+                };
+            });
+            #endregion
+        }
+
+        #region ServiceFabricPersistenceSagaWithCustomCollectionName
+        [ServiceFabricSaga(CollectionName = "custom-collection-name")]
+        public class CustomCollectionNameSaga : IHandleMessages<Message>
```

Comment:
This is a handler and not a saga.  Change to saga and make the region only show the declaration of the attribute with the class declaration?